test/makeTravisReport.m

```diff
@@ -1,45 +1,260 @@
+    % calculate the required column width
+    cellWidth   = cellfun(@length, [header(:)' ;data]);
+    columnWidth = max(max(cellWidth, [], 1),3); % use at least 3 places
+
+    % prepare the table format
+    COLSEP = '|'; ROWSEP = sprintf('\n');
+    rowformat = [COLSEP sprintf([' %%%ds ' COLSEP], columnWidth) ROWSEP];
+    alignmentRow = formatAlignment(alignment, columnWidth);
+
+    % actually print the table
+    fullTable = [header; alignmentRow; data];
+    strs = cell(size(fullTable,1), 1);
+    for iRow = 1:numel(strs)
+        thisRow = fullTable(iRow,:);
+        %TODO: maybe preprocess thisRow with strjust first
+        strs{iRow} = sprintf(rowformat, thisRow{:});
+    end
+    str = [strs{:}];
+    function alignRow = formatAlignment(alignment, columnWidth)
+        DASH = '-'; COLON = ':';
+        N = numel(columnWidth);
+        alignRow = arrayfun(@(w) repmat(DASH, 1, w), columnWidth, ...
+                        'UniformOutput', false);
+        for iColumn = 1:N
+            thisAlign = alignment(iColumn);
+            thisSpec = alignRow{iColumn};
+            switch lower(thisAlign)
+                case 'l'
+                    thisSpec(1) = COLON;
+                case 'r'
+                    thisSpec(end) = COLON;
+                case 'c'
+                    thisSpec([1 end]) = COLON;
+                otherwise
+                    error('gfmTable:BadAlignment','Unknown alignment "%s"',...
+                          thisAlign);
+            end
+            alignRow{iColumn} = thisSpec;
+        end
+    end
+end
+function str = gfmCode(str, inline, language)
+    % Constructs a GFM code fragment
+    %
+    % Arguments:
+    %   - str: code to be displayed
+    %   - inline:  - true  -> formats inline
+    %              - false -> formats as code block
+    %              - []    -> automatic mode (default): picks one of the above
+    %   - language: which language the code is (enforces a code block)
+    % 
+    % Output: GFM formatted string
+    if ~exist('inline','var')
+        inline = [];
+    end
+    if ~exist('language','var') || isempty(language)
+        language = '';
+    else
+        inline = false; % highlighting is not supported for inline code
+    end
+    if isempty(inline)
+        inline = isempty(strfind(str, sprintf('\n')));
+    end
+    
+    if inline
+        prefix = '`';
+        postfix = '`';
+    else
+        prefix = sprintf('\n```%s\n', language);
+        postfix = sprintf('\n```\n');
+        if str(end) == sprintf('\n')
+            postfix = postfix(2:end); % remove extra endline
+        end
+    end
+    
+    str = sprintf('%s%s%s', prefix, str, postfix);
+end
+function str = gfmHeader(str, level)
+    if ~exist('level','var')
+        level = 1;
+    end
+    str = sprintf('\n%s %s\n', repmat('#', 1, level), str);
+end
+function symbols = githubEmoji()
+    % defines the emojis to signal the test result
+    symbols = struct('pass', ':heavy_check_mark:', ...
```

Comment:
Could we use `:white_check_mark: ` :white_check_mark:  to make the color transport the message as well?